src/connectStyle.js

```diff
@@ -134,36 +174,25 @@ export default function connectStyle(
           // Usually they are set through alternative ways,
           // such as theme style, or through options
           addedProps: this.resolveAddedProps(),
-          styleNames,
+          styleNames, // eslint-disable-line
         };
       }
 
       getChildContext() {
+        const { virtual } = this.props;
+        const { childrenStyle } = this.state;
+        const { parentStyle } = this.context;
+
         return {
-          parentStyle: this.props.virtual ?
-            this.context.parentStyle :
-            this.state.childrenStyle,
+          parentStyle: virtual ? parentStyle : childrenStyle,
           transformProps: this.transformProps,
         };
       }
 
-      componentDidUpdate(prevProps) {
-        const styleNames = this.resolveStyleNames(this.props);
-
-        if (this.shouldRebuildStyle(prevProps, styleNames)) {
-          const resolvedStyle = this.resolveStyle(this.context, this.props, styleNames);
-
-          this.setState({
-            style: resolvedStyle.componentStyle,
-            childrenStyle: resolvedStyle.childrenStyle,
-            styleNames,
-          });
-        }
-      }
```

Comment:
Not sure about this.
Taking aside the fact that this code could be optimized, how do you know that the possibility of dynamic style change wasn't used up till now? (especially taking into account that this is a public lib).
Even if we wanted to remove this functionality, it shouldn't be hidden behind a minor version change and a code linting PR.

Note - `setState` can be used in `componentDidUpdate`, it just needs to be wrapped in a condition in order to avoid an endless loop (https://reactjs.org/docs/react-component.html#componentdidupdate).

Reply:
I was not aware of the condition wrapper allowing it.

That being said, the linter still states `react/no-did-update-set-state` rule is being broken when this is used, regardless of the condition wrapper.

This seems to actually be something that's [debated on eslint](https://github.com/yannickcr/eslint-plugin-react/issues/1707).

Should I reinstate it and add a `// eslint-disable-next-line react/no-did-update-set-state` comment?

Reply:
I would suggest that we exclude this rule altogether. You can exclude it for now in `package.json` (eslintConfig part you added), but I think we should actually exclude it from our shoutem react-eslint plugin.

Reply:
Will disable rule via `package.json`, also created [PR for this on our repo](https://github.com/shoutem/eslint-config/pull/2).